### PR TITLE
Add steel armor clothing mod

### PR DESCRIPTION
**Pull request by snipercup**
#### Summary
```SUMMARY: Content "Adds steel armor clothing mod"```

#### Purpose of change
Follow up on #34571. 

#### Describe the solution
You can now pad your clothes with steel_armor and a tailors kit. 

#### Describe alternatives you've considered
I used the same stats as leather_padded but i feel like there should be a difference. maybe have leather_padded add warmth?

#### Additional context
image

image

### CODE REVIEW

**Comment by tenmillimaster** [comment]
Should the steel padded armor variants get obsoleted then?

Also if call it 'lined', not 'padded'.

**Comment by snipercup** [comment]
Can you give me an example of a steel padded armor variant?

**Comment by LaVeyanFiend** [comment]
> Can you give me an example of a steel padded armor variant?

Armored leather jacket or armored leather vest

**Comment by Rivet-the-Zombie** [comment]
>pair of armored gauntlets, pair of armored boots

These two aren't just 'a bit of clothing with added metal' but are rather meant to be armor in their own right - the non-metal bits just serve to keep the metal pieces together.

**Comment by snipercup** [comment]
I'm scratching my head over the plated leather armor but I think i'll come up with something and hopefully I can extend that to the other items you mentioned.

**Comment by DaviBones** [comment]
I am willing to take a stab at the C++ legwork, if you come up with a JSON flag/property you want to use to differentiate it.

**Comment by snipercup** [comment]
> 
> 
> I am willing to take a stab at the C++ legwork, if you come up with a JSON flag/property you want to use to differentiate it.

Thanks for wanting to help. The first thing that comes up is a whitelist flag that can be applied to any clothing that you want to be able to be padded. For example when entering an item's flags, you add can_pad_steel, can_pad_leather, can_line_fur

Alternatively you could check an itemgroup and items in that group can be padded

**Comment by DaviBones** [comment]
How about using something existing, like `material_thickness`?

**Comment by snipercup** [comment]
I will look into it.

**Comment by snipercup** [comment]
I think a pretty good selection is items that have material: leather, flag: sturdy and material_thickness: >3 and only Covers: Legs and/or torso (and arms). I can probably write a query to see what items come up but I'm pretty confident that these items are a good selection to start with. This would also exclude the items LaVeyanFiend mentions except for the plated leather armor. I can think of some options for the plated leather armor but i'll get to that later.

**Comment by DaviBones** [comment]
Just thought of a more elegant way of doing it, sort of an extension of one of your ideas: We'll have an optional JSON property `valid_items` for clothing mods which, if present, restricts the clothing mod such that it can only be applied to the listed items.

@kevingranade Is this an acceptable solution?

**Comment by kevingranade** [comment]
That's better than the heuristic approach.
Generally you'd flip it around and put it on the clothing, i.e. ```"valid_mods": [ "steel_insert" ],``.

**Comment by DaviBones** [comment]
What's the justification for doing it that way instead? I kind of prefer my approach, as it doesn't invalidate the existing four clothing mods until someone goes through all the existing armor/clothing and adds `valid_mods` entries where applicable.

**Comment by kevingranade** [comment]
Neither one invalidates anything unless you code it to do so.  Treat it as a whitelist and everything is fine.

As for the reason, it's far easier to maintain tags on items than it is to maintain centralized lists of items. Similarly it handles new items added by mods more seamlessly.

**Comment by snipercup** [comment]
@kevingranade Can this PR be merged now that #34705 is merged? I will make another PR that will put this steel clothing mod to use.

**Comment by snipercup** [comment]
What is holding this PR back from being merged?

**Comment by cainiaowu** [comment]
mostly lack of maintainers time, you may need to ask directly on discord or it may get missed.

**Comment by snipercup** [comment]
@kevingranade Could you take another look at this PR?

File: data/json/clothing_mods.json [hunk 1]
```diff
@@ -12,6 +12,19 @@
       { "type": "encumbrance", "value": 2, "round_up": true, "proportion": [ "thickness", "coverage" ] }
     ]
   },
+  {
+    "type": "clothing_mod",
+    "id": "steel_padded",
+    "flag": "steel_padded",
+    "item": "steel_armor",
+    "implement_prompt": "Pad with steel",
+    "destroy_prompt": "Destroy steel padding",
```
Comment (kevingranade) [inline data/json/clothing_mods.json]:
```suggestion
    "destroy_prompt": "Destroy steel padding",
    "restricted": true,
```

**Comment by kevingranade** [comment]
It needs to be marked restricted as discussed, and have at least one article of clothing with it added to the modification whitelist.

**Comment by snipercup** [comment]
Steel padded clothing mod is now restricted.
I added valid_mods to some clothing items.
Steel padded can now be applied:
image


Cannot be applied to other clothing:
image


Please review/merge

**Pull request merged**